src/compiler/factoryPublic.ts

```diff
@@ -2270,6 +2270,18 @@ namespace ts {
             : node;
     }
 
+    export function createDynamicImport(argumentsArray: readonly Expression[], typeArguments: readonly TypeNode[] = []) {

```

Comment:
If the `importKeyword` is undefined, should I give a default value for it?

Reply:
Yes, the point of having `undefined` be allowed is so that you can pass `undefined` in and let ts create the node for you if you don't need it. `undefined` would *not* be allowed in the `update` function though. We have a few other factory functions that are similar to this.